cpp/src/io/statistics/byte_array_view.cuh

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+namespace cudf::io {
+
+/**
+ * @brief Wrapper for a row of a list<int8> or list<uint8> column. This is analagous to
+ * `string_view` in type. It was created due to the need for comparison operators for cub reduce on
+ * statistics. Otherwise, it is a device_span in all but name.
+ *
+ */
+class byte_array_view {
+ public:
+  using element_type = uint8_t const;  ///< The type of the elements in the byte array
```

Comment:
Can we use `std::byte` here?

Reply:
Sorry, had to revert this. It doesn't work with a `column_device_view` due to SFINAE'd data function:

```
  template <typename T, CUDF_ENABLE_IF(is_rep_layout_compatible<T>())>
  [[nodiscard]] CUDF_HOST_DEVICE T const* data() const noexcept
```
`is_rep_layout_compatible` is
```
template <typename T>
constexpr bool is_rep_layout_compatible()
{
  return cudf::is_numeric<T>() or cudf::is_chrono<T>() or cudf::is_boolean<T>();
}
```

This doesn't include `std::byte`. I'm happy to pursue this later, but for now I would rather see it run with `uint8_t`. Should I add an issue?

Reply:
I think have an issue would be helpful to std::byte adoption, thank you.

Reply:
https://github.com/rapidsai/cudf/issues/11362